src/Data/Avro/Encoding/FromAvro.hs

```diff
@@ -316,21 +317,21 @@ getBlocksOf env t = do
     (vs:) <$> getBlocksOf env t
 
 writeByPositions :: MV.MVector s Value -> [(Int, Value)] -> ST s ()
-writeByPositions mv writes = foldl (>>) (return ()) (fmap (go mv) writes)
+writeByPositions mv = traverse_ (go mv)
   where go :: MV.MVector s Value ->  (Int, Value) -> ST s ()
-        go mv (n, v) = MV.write mv n v
+        go mv = uncurry (MV.write mv)
 
 getRecord :: HashMap Schema.TypeName ReadSchema -> [ReadSchema.ReadField] -> Get (Vector Value)
 getRecord env fs = do
-  moos <- forM fs $ \f ->
+  moos <- fmap concat . forM fs $ \f ->
     case ReadSchema.fldStatus f of
-      ReadSchema.Ignored       -> getField env (ReadSchema.fldType f) >> pure []
-      ReadSchema.AsIs i        -> fmap ((:[]) . (i, )) (getField env (ReadSchema.fldType f))
+      ReadSchema.Ignored       -> [] <$ getField env (ReadSchema.fldType f)
+      ReadSchema.AsIs i        -> (\f -> [(i,f)]) <$> getField env (ReadSchema.fldType f)
       ReadSchema.Defaulted i v -> pure [(i, convertValue v)] --undefined
 
   return $ V.create $ do
-    vals <- MV.unsafeNew (length fs)
```

Comment:
Specifically the bug was here. Length fs isn't correct, it should be the same as the length below. 
Causes an exection
`Data.Vector.Mutable: uninitialised element.` with no stack trace or source location.